avocado/core/loader.py

```diff
@@ -44,6 +44,41 @@
 ALL = True
 
 
+def filter_test_tags(test_suite, filter_by_tags):
+    '''
+    The filtering mechanism is limited to instrumented tests with tags set
+    '''
+    filtered = []
+    for klass, info in test_suite:
+        if not isinstance(klass, str):  # not instrumented: no filtering
+            filtered.append((klass, info))
+            continue
+
+        test_tags = info['tags']
+        if not test_tags:       # not tagged: no filtering
```

Comment:
Actually not, this would have to be `if test_tags is None` and the loader would have to initialize it as `None`, or set of tags (including `set()`).